src/js/tabs/pid_tuning.js

```diff
@@ -1749,13 +1765,21 @@ pid_tuning.initialize = function (callback) {
             const midyl = canvasHeight - ((canvasHeight - midy) * 0.5 *(expo + 1));
             const midyr = (midy / 2) * (expo + 1);
 
-            let thrPercent = (FC.RC.channels[3] - 1000) / 1000,
-                thrpos = thrPercent <= mid
-                    ? getQuadraticCurvePoint(0, canvasHeight, midxl, midyl, midx, midy, thrPercent * (1.0 / mid))
-                    : getQuadraticCurvePoint(midx, midy, midxr, midyr, canvasWidth, 0, (thrPercent - mid) * (1.0 / (1.0 - mid)));
+            const thrPercent = (FC.RC.channels[3] - 1000) / 1000;
+            const thrpos = calculateThrottlePosition(thrPercent, canvasWidth, canvasHeight, mid, midx, midxl, midxr, midy, midyr, midyl);
+            let thrposLimit = null;
+            if (throttleLimitType === THROTTLE_LIMIT_TYPES.CLIP) {
+                thrposLimit = calculateThrottlePosition(throttleLimitPercent, canvasWidth, canvasHeight, mid, midx, midxl, midxr, midy, midyr, midyl);
+
+                if (throttleLimitPercent < thrPercent) {
```

Comment:
`if (thrpos.y < thrposLimit.y) {`

Reply:
I think this may have the desired effect.

Reply:
No, the problem is more complicated... this `thrposLimit` is calculated using what I thought is max throttle allowed, but it limits the "input" stick, not the output produced. So this is not valid. I need to work using the total Y "canvas" size to limit the output and show the current position of the throttle.
I will work on it...

Reply:
@McGiverGim : Usually, it is better to ignore canvas size during curve calculation - simply generate curve width = 0.0 .. 1.0, height = 0.0..1.0 (or -1..1 when centered / throttle values from BF is simpler). And scale it to canvas in last operation.

Also, it may be possible to copy (all) throttle code from BF and just transliterate it from C to javascript ...

Reply:
@ledvinap thanks for your comment. In the SCALE method is what I do, generate the curve and reduce the canvas size. But in the CLIP is not as easy. There is a function that can produce the curve, but of course, it has no upper limit. It's easy to limit "the current" value but not as easy to limit the curve starting at some point.
The code from firmware calculates the "standard" value and if CLIP is enabled, it limits it to the configured value, but the firmware does not need to draw the total curve.

@ChrisRosser I've uploaded a new version. I think this shows the desired result but I don't like the code. Let's see if someone has a suggestion about how to make it better.